### PR TITLE
Gracefully handle the not_built message from the hashtree.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Ensure that when aae_source dies, we handle the DOWN message and
terminate the fsssource process instead of ignoring the message.

Do not throw the error, or send the complete message to the sink, when
the process dies; both are not necessary.

### CODE REVIEW

**Comment by buddhisthead** [comment]
Hey Chris, what test do you recommend to reproduce this? Will repl_aae_fullsync show it, if I reduce the max source to 1?

File: src/riak_repl2_fssource.erl [hunk 1]
```diff
@@ -182,6 +182,10 @@ handle_info({'DOWN', Ref, process, _Pid, not_responsible}, State=#state{partitio
         {ok, State2} -> {noreply, State2};
         Error -> Error
     end;
+handle_info({'DOWN', Ref, process, _Pid, Reason}, State) ->
+    erlang:demonitor(Ref),
```
Comment (Vagabond) [inline src/riak_repl2_fssource.erl]: You don't need to demonitor a dead process.

File: src/riak_repl2_fssource.erl [hunk 2]
```diff
@@ -182,6 +182,10 @@ handle_info({'DOWN', Ref, process, _Pid, not_responsible}, State=#state{partitio
...
+    lager:info("Received: ~p, stopping process.", [Reason]),
```
Comment (Vagabond) [inline src/riak_repl2_fssource.erl]: If the reason is Error or shutdown, we can probably suppress this.

Comment (Vagabond) [inline src/riak_repl2_fssource.erl]: normal or shutdown, I mean.

File: src/riak_repl_aae_source.erl [hunk 3]
```diff
@@ -170,8 +170,6 @@ prepare_exchange(start_exchange, State=#state{transport=Transport,
         Error ->
             lager:info("AAE source failed get_lock for partition ~p, got ~p",
                        [Partition, Error]),
-            send_complete(State),
```
Comment (Vagabond) [inline src/riak_repl_aae_source.erl:173]: We should also remove the other throw()s like this, they will all trigger bad_return_type errors. We should also audit the other send_completes to see if they make sense vs just having the socket close.

**Comment by cmeiklejohn** [comment]
Closing, incorporated into #500.

**Pull request closed without merging**